File: models/comment_reply.go
package models

import "time"

type CommentReply struct {
	ID        int       `json:"id" gorm:"primaryKey"`
	UserID    int       `json:"user_id" gorm:"not null"`
	CommentID int       `json:"comment_id" gorm:"not null"`
	Message   string    `json:"message" gorm:"size:200;not null"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
	User      User      `json:"user"`
	Comment   Comment   `json:"comment"`
}

File: web/haiwangram/src/interface/CommentReply.ts
// {
//     "id": 1,
//     "message": "lorem ipsum dolor sit amet consectetur adipiscing elit condimentum duis",
//     "comment_id": 1,
//     "user_id": 3,
//     "comment": {
//       "id": 1,
//       "message": "lorem ipsum dolor sit amet consectetur adipiscing",
//       "photo_id": 1,
//       "user_id": 3
//     },
//     "user": {
//       "id": 3,
//       "email": "sqnewm-hkykbr@example.com",
//       "username": "sqnewm-hkykbr",
//       "full_name": "Sqnewm Hkykbr",
//       "profile_image_url": "https://randomuser.me/api/portraits/women/87.jpg",
//       "description": "lorem ipsum dolor sit amet consectetur adipiscing"
//     }
//   }

import Comment from "./Comment";
import User from "./User";

interface CommentReply {
  id: number;
  message: string;
  comment_id: number;
  user_id: number;
  comment: Comment;
  user: User;
}

export default CommentReply;
